service/worker/archiver/workflow_test.go

```diff
@@ -104,34 +118,58 @@ func (s *workflowSuite) TestArchivalWorkflow_Exit_TimeoutWithoutSignals() {
+	env.ExecuteWorkflow(archiveHistoryWorkflowTest)
 
 	s.True(env.IsWorkflowCompleted())
 	s.NoError(env.GetWorkflowError())
 	env.AssertExpectations(s.T())
 }
 
-func (s *workflowSuite) TestArchivalWorkflow_Success() {
-	workflowTestMetrics.On("IncCounter", metrics.ArchiverArchivalWorkflowScope, metrics.ArchiverWorkflowStartedCount).Once()
-	workflowTestMetrics.On("StartTimer", metrics.ArchiverArchivalWorkflowScope, metrics.CadenceLatency).Return(metrics.NopStopwatch()).Once()
-	workflowTestMetrics.On("StartTimer", metrics.ArchiverArchivalWorkflowScope, metrics.ArchiverHandleAllRequestsLatency).Return(metrics.NopStopwatch()).Once()
-	workflowTestMetrics.On("AddCounter", metrics.ArchiverArchivalWorkflowScope, metrics.ArchiverNumPumpedRequestsCount, int64(5)).Once()
-	workflowTestMetrics.On("AddCounter", metrics.ArchiverArchivalWorkflowScope, metrics.ArchiverNumHandledRequestsCount, int64(5)).Once()
+func (s *workflowSuite) TestArchiveHistoryWorkflow_Success() {
+	workflowTestMetricsClient.On("Scope", metrics.HistoryArchivalWorkflowScope, mock.Anything).Return(workflowTestMetricsScope).Once()
+	workflowTestMetricsScope.On("IncCounter", metrics.ArchiverWorkflowStartedCount).Once()
+	workflowTestMetricsScope.On("StartTimer", metrics.CadenceLatency).Return(metrics.NewTestStopwatch()).Once()
+	workflowTestMetricsScope.On("StartTimer", metrics.ArchiverHandleAllRequestsLatency).Return(metrics.NewTestStopwatch()).Once()
+	workflowTestMetricsScope.On("AddCounter", metrics.ArchiverNumPumpedRequestsCount, int64(5)).Once()
+	workflowTestMetricsScope.On("AddCounter", metrics.ArchiverNumHandledRequestsCount, int64(5)).Once()
 	workflowTestHandler.On("Start").Once()
 	workflowTestHandler.On("Finished").Return([]uint64{1, 2, 3, 4, 5}).Once()
 	workflowTestPump.On("Run").Return(PumpResult{
 		PumpedHashes: []uint64{1, 2, 3, 4, 5},
 	}).Once()
 
 	env := s.NewTestWorkflowEnvironment()
-	env.ExecuteWorkflow(archivalWorkflowTest)
+	env.ExecuteWorkflow(archiveHistoryWorkflowTest)
 
 	s.True(env.IsWorkflowCompleted())
 	_, ok := env.GetWorkflowError().(*workflow.ContinueAsNewError)
 	s.True(ok, "Called ContinueAsNew")
 	env.AssertExpectations(s.T())
 }
 
-func archivalWorkflowTest(ctx workflow.Context) error {
-	return archivalWorkflowHelper(ctx, workflowTestLogger, workflowTestMetrics, workflowTestConfig, workflowTestHandler, workflowTestPump, nil)
+func (s *workflowSuite) TestReplayArchiveHistoryWorkflow() {
+	logger, _ := zap.NewDevelopment()
+	globalLogger = workflowTestLogger
+	globalMetricsClient = metrics.NewClient(tally.NewTestScope("replay", nil), metrics.Worker)
+	globalConfig = &Config{
+		ArchiverConcurrency:           dynamicconfig.GetIntPropertyFn(50),
+		ArchivalsPerIteration:         dynamicconfig.GetIntPropertyFn(1000),
+		TimeLimitPerArchivalIteration: dynamicconfig.GetDurationPropertyFn(MaxArchivalIterationTimeout()),
+	}
+	err := worker.ReplayWorkflowHistoryFromJSONFile(logger, "testdata/archive_history_workflow_history.json")
```

Comment:
Can you chat with Maxim about how these files should be named? It seems like each time we make a workflow change we want to add a history to test and those histories need some nice name?